specification/notificationhubs/Notificationhubs.Management/models.tsp

```diff
@@ -0,0 +1,1600 @@
+   */
+  @visibility(Lifecycle.Read)
+  groupId?: string;
+
+  /**
+   * Required members. For Notification Hubs, it's always a collection with a single "namespace" item.
+   */
+  @visibility(Lifecycle.Read)
+  requiredMembers?: string[];
+
+  /**
+   * Required DNS zone names. For Notification Hubs, it contains two CNames for Service Bus and Notification Hubs zones.
+   */
+  @visibility(Lifecycle.Read)
+  requiredZoneNames?: string[];
+}
+
+/**
+ * Part of Private Endpoint description that stores information about a connection between Private Endpoint and Notification Hubs namespace.
+ * This is internal class, not visible to customers, and we use it only to discover the link identifier.
+ */
+model ConnectionDetails {
+  /**
+   * A unique ID of the connection. This is not the ARM id, but rather an internal identifier set by the Networking RP. Notification Hubs code
+   * does not analyze it.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+
+  /**
+   * IP address of the Private Endpoint. This is not used by Notification Hubs.
+   */
+  @visibility(Lifecycle.Read)
+  privateIpAddress?: string;
+
+  /**
+   * Link identifier. This is a string representation of an integer that is also encoded in every IPv6 frame received by Front Door,
+   * and we use it to create implicit authorization rule that allows connection from the associated Private Endpoint.
+   */
+  @visibility(Lifecycle.Read)
+  linkIdentifier?: string;
+
+  /**
+   * Group name. Always "namespace" for Notification Hubs.
+   */
+  @visibility(Lifecycle.Read)
+  groupId?: string;
+
+  /**
+   * Member name. Always "namespace" for Notification Hubs.
+   */
+  @visibility(Lifecycle.Read)
+  memberName?: string;
+}
+
+/**
+ * Represents a connectivity information to Notification Hubs namespace. This is part of PrivateLinkService proxy that tell
+ * the Networking RP how to connect to the Notification Hubs namespace.
+ */
+model GroupConnectivityInformation {
+  /**
+   * Group id. Always set to "namespace".
+   */
+  @visibility(Lifecycle.Read)
+  groupId?: string;
+
+  /**
+   * Member name. Always set to "namespace".
+   */
+  @visibility(Lifecycle.Read)
+  memberName?: string;
+
+  /**
+   * List of customer-visible domain names that point to a Notification Hubs namespace.
+   */
+  @visibility(Lifecycle.Read)
+  customerVisibleFqdns?: string[];
+
+  /**
+   * One of the domain name from the customer-visible names; this is used internally by Private Link service to make connection to Notification Hubs
+   * namespace.
+   */
+  @visibility(Lifecycle.Read)
+  internalFqdn?: string;
+
+  /**
+   * Not used by Notification Hubs.
+   */
+  @visibility(Lifecycle.Read)
+  redirectMapId?: string;
+
+  /**
+   * ARM region for Private Link Service. We use the region that contains the connected Notification Hubs namespace.
+   */
+  @visibility(Lifecycle.Read)
+  privateLinkServiceArmRegion?: string;
+}
```

Comment:
This comment suggests its an internal API contract, in which case it shouldn't even be documented here should it?

>
/**
 * Represents a connectivity information to Notification Hubs namespace. This is part of PrivateLinkService proxy that tell
 * the Networking RP how to connect to the Notification Hubs namespace.
 */